paasta_tools/paasta_cli/cmds/status.py

```diff
@@ -125,69 +130,76 @@ def get_actual_deployments(service):
     return actual_deployments
 
 
-def report_status_for_cluster(service, cluster, deploy_pipeline, actual_deployments, verbose=False):
+def report_status_for_cluster(service, cluster, deploy_pipeline, actual_deployments, instance_whitelist, verbose=False):
     """With a given service and cluster, prints the status of the instances
     in that cluster"""
     # Get cluster.instance in the order in which they appear in deploy.yaml
     print
     print "cluster: %s" % cluster
+    seen_instances = []
     for namespace in deploy_pipeline:
         cluster_in_pipeline, instance = namespace.split('.')
+        seen_instances.append(instance)
 
         if cluster_in_pipeline != cluster:
-            # This function only prints things that are relevant to cluster
-            # We skip anything not in this cluster
+            continue
+        if instance_whitelist and instance not in instance_whitelist:
             continue
 
         # Case: service deployed to cluster.instance
         if namespace in actual_deployments:
-            unformatted_instance = instance
-            instance = PaastaColors.blue(instance)
+            formatted_instance = PaastaColors.blue(instance)
             version = actual_deployments[namespace][:8]
             # TODO: Perform sanity checks once per cluster instead of for each namespace
-            status = execute_paasta_serviceinit_on_remote_master('status', cluster, service, unformatted_instance,
-                                                                 verbose=verbose)
+            status = execute_paasta_serviceinit_on_remote_master('status', cluster, service, instance, verbose=verbose)
         # Case: service NOT deployed to cluster.instance
         else:
-            instance = PaastaColors.red(instance)
+            formatted_instance = PaastaColors.red(instance)
             version = 'None'
             status = None
 
-        print '  instance: %s' % instance
+        print '  instance: %s' % formatted_instance
         print '    Git sha:    %s' % version
         if status is not None:
             for line in status.rstrip().split('\n'):
                 print '    %s' % line
 
+    print report_bogus_whitelists(instance_whitelist, seen_instances, 'instance')
 
-def report_bogus_filters(cluster_filter, deployed_clusters):
-    """Warns the user if the filter used is not even in the deployed
+
+def report_bogus_whitelists(whitelists, items, item_type):
+    """Warns the user if the whitelist used is not even in the deployed
```

Comment:
this docstring doesn't make sense to me. """Warns the user if there are invalid values in the whitelist""" would be clearer.